pkg/operator/metriccontroller/fsync_controller.go

```diff
@@ -105,7 +117,37 @@ func (c *FSyncController) sync(ctx context.Context, syncCtx factory.SyncContext)
 	// Send warning event if excessive leader changes are detected. The event will include fsync disk metrics.
 	syncCtx.Recorder().Warningf("EtcdLeaderChangeMetrics", "Detected leader change increase of %s over 5 minutes on %q; disk metrics are: %s. Most often this is as a result of inadequate storage or sometimes due to networking issues.", leaderChanges, platformType, strings.Join(values, ","))
 
-	// TODO: Consider Degraded condition here.
+	// Capture any etcd alerts. We want to go degraded if critical etcd alerts are firing to warn the admin.
+	etcdAlertsResult, _, err := client.Query(ctx, "ALERTS{alertname=~\"etcd.+\", job=\"etcd\"}", time.Now())
```

Comment:
If there is a bug or corner case with alerts I think we risk upgrade disruption.

Reply:
Yes agreed, we can also skip going degraded on alerts for now until we stabilize the alerts. I can open a follow up jira ticket for the future releases? And for now only go degraded on leader changes and fsync?

Reply:
sgtm, thanks!

Reply:
Done, also added TODO in the code with the query that would be used. https://issues.redhat.com/browse/ETCD-239